### DISCUSSION TITLE
Create new note, move, then place the cursor at the end of the file

### DISCUSSION THREAD

**Discussion by gpanakkal** [SilentVoid13/Templater#1176] [Help] [answered]
Hi, very new to templater here. I made this template:

```
<%*
const newFileName = tp.date.now("YYYY-MMM-DD  hh-mm-ss ZZ");
await tp.file.move("/Inbox/" + newFileName);
tp.file.cursor();
%>
```

When I run "create new note from template" it correctly moves and names the file, but then the title becomes highlighted. Since the template also has some frontmatter, I have to hit tab and then page down to get to the point where I want the cursor to be initially. I've already enabled automatic jump to cursor and restarted Obsidian. The cursor() code disappears, indicating it has run, even when it's in its own <% %> block. How do I fix this?

**Comment by Zachatoo** [comment]
There's not a way around having to hit the `tab` key when creating new files. However, on hitting `tab`, it should take you to your cursor. Here's an example template that works for me.

```js
<%*
const newFileName = tp.date.now("YYYY-MMM-DD  hh-mm-ss ZZ");
await tp.file.move("/Inbox/" + newFileName);
-%>
<% "---" %>
key: value
<% "---" %>

# <% newFileName %>
<% tp.file.cursor() %>
```

**Reply by anamyk** [reply]
Was looking for this information. Thanks for claryfying.

Quick question, I found a snippet on [reddit](https://www.reddit.com/r/ObsidianMD/comments/10d0qv1/am_i_losing_my_mind_or_does_tpfilecursor_just/) with this command
`<% tp.file.cursor() %><%* app.workspace.activeLeaf.view.editor?.focus(); %>`

but this also does not get rid of the TAB with new file creation by Templater, right? 
I couldn't get it to work.

**Reply by Zachatoo** [reply]
That code works for me when using folder templates, but not when using the "create new note from template" command.

I've modified it to wait to focus the editor until after Templater has finished execution. The hooks block can go anywhere in your template.

```js
<% tp.file.cursor() %>
<%*
tp.hooks.on_all_templates_executed(() => {
  app.workspace.activeLeaf.view.editor?.focus();
});
-%>
```

**Reply by bradleyjay** [reply]
This is amazing! Any idea how you would modify this to place the cursor at the beginning of the title, instead of in the body? @Zachatoo

**Reply by Zachatoo** [reply]
Can you clarify what you mean by title? A screenshot would be helpful.

**Reply by bradleyjay** [reply]
Thanks! For sure. I mean, in the first line of the note, _not_ the body. I.e. 

<img width="536" alt="image" src="https://github.com/user-attachments/assets/9e28ba3b-0480-41cc-b86a-b4332a4abb06" />
![note_example_screenshot](https://github.com/user-attachments/assets/ffcc0e5b-06c5-4d5a-bfe8-c28b3f4465ab)

Doesn't seem like code snippets work in the title itself, so any tp.cursor call there gets ignored. However, putting it in the body DOES work, just like you said.

**Reply by Zachatoo** [reply]
I don't believe you can do that with Templater, no. Shouldn't it do that by default when creating a note?

**Comment by Zachatoo** [comment, marked as answer]
I'm adding a "Jump to note content after creating note" setting, when enabled, will jump your cursor to the note content without having to tab into it.

<img width="888" height="776" alt="Screenshot 2026-06-05 at 10 05 48 PM" src="https://github.com/user-attachments/assets/5a5b7074-66b9-4f63-b266-d272141d2d7e" />

This is in the Templater 2.22.0 release, which is only available on Obsidian 1.13.0+
